utilities/test_suite/HIP/Tensor_hip.cpp

```diff
@@ -610,6 +614,88 @@ int main(int argc, char **argv)
 
                 break;
             }
+            case 82:
+            {
+                testCaseName = "ricap";
+                double randFromDist, randFromDist1;
+
+                if(!qaFlag)
+                {
+                    float betaParam = 0.3;
+                    std::random_device rd;
+                    std::mt19937 gen(rd());
+                    static std::uniform_real_distribution<double> unif(0.3, 0.7);
+                    double p = unif(gen);
+                    randFromDist = boost::math::ibeta_inv(betaParam, betaParam, p);
+
+                    std::random_device rd1;
+                    std::mt19937 gen1(rd1());
+                    static std::uniform_real_distribution<double> unif1(0.3, 0.7);
+                    double p1 = unif1(gen1);
+                    randFromDist1 = boost::math::ibeta_inv(betaParam, betaParam, p1);
+                }
+
+                int iX = maxWidth;
+                int iY = maxHeight;
+
+                Rpp32u initialPermuteArray[batchSize], permutedArray[batchSize * 4], permutationTensor[batchSize * 4];
+                for (uint i = 0; i < batchSize; i++)
+                {
+                    initialPermuteArray[i] = i;
+                }
+
+                if(qaFlag)
```

Comment:
@snehaa8 this case 82 part of the code is still written very inefficiently, we are checking if(qaFlag) three times while we could check just once by rearrranging code. Also try to see if changing data layout somewhere else can reduce work in all these memcpy calls and perrmutationTensor arrangement. We can have a call after you take a look tomorrow

Reply:
Sure I will combine parts that get enabled for qa together
Reduced checking if qa to 2 times.

I don't think changing data layout will help reduce memcpy and perrmutationTensor arrangement. As batchsize can be anything but we just support 4 partitions in any given image.